library/std/src/sys/unix/net.rs

```diff
@@ -443,6 +443,21 @@ impl Socket {
         self.0.set_nonblocking(nonblocking)
     }
 
+    #[cfg(target_os = "linux")]
+    pub fn set_mark(&self, mark: u32) -> io::Result<()> {
+        setsockopt(self, libc::SOL_SOCKET, libc::SO_MARK, mark as libc::c_int)
+    }
+
+    #[cfg(target_os = "freebsd")]
+    pub fn set_mark(&self, mark: u32) -> io::Result<()> {
+        setsockopt(self, libc::SOL_SOCKET, libc::SO_USER_COOKIE, mark)
+    }
+
+    #[cfg(target_os = "openbsd")]
+    pub fn set_mark(&self, mark: u32) -> io::Result<()> {
+        setsockopt(self, libc::SOL_SOCKET, libc::SO_RTABLE, mark as libc::c_int)
+    }
```

Comment:
The signature is identical across all of these platforms, and the implementation is nearly identical. How about deduplicating it this way?&mdash;

```rust
pub fn set_mark(&self, mark: u32) -> io::Result<()> {
    #[cfg(target_os = "linux")]
    let option = libc::SO_MARK;
    #[cfg(target_os = "linux")]
    let option = libc::SO_USER_COOKIE;
    #[cfg(target_os = "linux")]
    let option = libc::SO_RTABLE;
    setsockopt(self, libc::SOL_SOCKET, option, mark as libc::c_int)
}
```

or maybe this:

```rust
pub fn set_mark(&self, mark: u32) -> io::Result<()> {
    setsockopt(
        self,
        libc::SOL_SOCKET,
        #[cfg(target_os = "linux")]
        libc::SO_MARK,
        #[cfg(target_os = "freebsd")]
        libc::SO_USER_COOKIE,
        #[cfg(target_os = "openbsd")]
        libc::SO_RTABLE,
        mark as libc::c_int,
    )
}
```